types/index.test-d.ts

```diff
@@ -153,7 +153,7 @@ expectType<string[]>(tags);
 expectType<Credentials>(credentials);
 
 // Ensure duplicayed simple-oauth2 are compatible with simple-oauth2
-expectAssignable<ModuleOptions<string>>(credentials);
+expectAssignable<ModuleOptions<string>>({auth: {tokenHost: ''}, ...credentials });
```

Comment:
You aren't testing the case where it's `undefined` here?

Reply:
@gurgunday  This test, from what I understand, is comparing the interfaces with the `simple-oauth2` library, in this case the interface is "temporarily incompatible" at runtime when using `discovery` the `auth` value is configured, otherwise it fails.

I believe it has no impact, and tests for undefined `auth` already exist in the `discovery` suite.

Any suggestion I can modify!

Reply:
@mcollina do you have any suggestions for improvement?

Reply:
@mcollina  @gurgunday i write one more assertion in runtime, need something else?

`options.discovery.issuer or credentials.auth have to be given`